### ISSUE TITLE
Feature request: Ability to insert R code snippets

### ISSUE THREAD

**Issue by drleehw** [jasp-stats/jasp-desktop#2289]
The pace of development with JASP is very impressive, and it's now to the point where it's usable for many analyses, but there will always remain a great many more specialised algorithms one might wish to run. In most packages, e.g. SPSS, this can be done in syntax, so that the analysis can all be run in one environment, and the output is all together in one place. 

Could JASP please have the ability to put add R code snippets to the output, and have them execute? I see it working something like this:

1. Press the R Code Snippet button on the menu.
2. A blank output space appears on the right, and an R scripting window appears on the left. 
3. User types or pastes in some R code on the left.
4. JASP executes the code much as RStudio does, and displays all output, text and graphical, in the given space on the right. 
5. The code snippet remains editable if one clicks back on the relevant section in the output. 

This could also be a nice solution for manipulating variables for power users so long as R code snippets in R were had the ability to alter data files. 

I know it's a big request, but it would be a big step in JASP getting to the point where I could use it to teach postgrad stats. I'm sure many users would be very very grateful.

**Comment by EJWagenmakers** [comment]
Yes, this is another great suggestion. We have just implemented something like this for a subset of options (for the version following the upcoming version); doing the entire analysis as you suggest will be a big ask but we are working our way towards it.

**Comment by boutinb** [comment]
We are working on 2 features where R scripts will be directly available for users: for filtering data, and for computed columns. These features should be available over 2 months.
We are also working of making the results available directly in R: JASP will be able to generate R scripts that will be directly executable in R studio, and that will produce the same output as in JASP. You will have then all opportunities to modify this generated R code.
Also, our plan is to allow R programmeur to be able to add easily (and dynamically) new analyses in JASP. This should be available over half a year.
